pkg/sql/backfill/backfill.go

```diff
@@ -672,13 +699,27 @@ func (ib *IndexBackfiller) BuildIndexEntriesChunk(
 		); err != nil {
 			return nil, nil, err
 		}
+		// The index keys can be large and so we must account for them in the index
+		// backfiller memory monitor.
+		for _, index := range buffer {
+			if err := ib.boundAccount.Grow(ctx, int64(len(index.Key))); err != nil {
+				return nil, nil, err
+			}
+		}
+
+		// There can be a large number of index KVs (egs: in the case of an inverted
+		// index), thus the size of the buffer must also be accounted for in the
+		// index backfiller memory monitor.
+		if err := ib.boundAccount.Grow(ctx, int64(unsafe.Sizeof(buffer))); err != nil {
```

Comment:
It'd really be better to do this growing before you create the `buffer`. You in general want to grow the account and then allocate the memory if no errors.